scripts/import-test-connections.js

```diff
@@ -0,0 +1,277 @@
+  connections.push({
+    favorite: {
+      name: 'Atlas'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_ATLAS_URL,
+    }
+  })
+}
+
+
+if (COMPASS_TEST_FREE_TIER_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas (free tier)'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_FREE_TIER_URL,
+    }
+  })
+}
+
+if (COMPASS_TEST_SECONDARY_NODE_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas (ReadPreference=Secondary)'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_SECONDARY_NODE_URL,
+    }
+  })
+}
+
+if (COMPASS_TEST_ANALYTICS_NODE_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas (nodeType:ANALYTICS)'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_ANALYTICS_NODE_URL,
+    }
+  })
+}
+
+if (E2E_TESTS_ATLAS_HOST && E2E_TESTS_ATLAS_X509_PEM_PATH) {
+
+  const url = new ConnectionStringUrl(
+    `mongodb+srv://${E2E_TESTS_ATLAS_HOST || ''}/admin`
+  );
+
+  url.searchParams.set('authMechanism', 'MONGODB-X509');
+  url.searchParams.set('tls', 'true');
+  url.searchParams.set('tlsCertificateKeyFile', E2E_TESTS_ATLAS_X509_PEM_PATH);
+  url.searchParams.set('authSource', '$external');
+
+  connections.push({
+    favorite: {
+      name: 'Atlas (X509)'
+    },
+    connectionOptions: {
+      connectionString: url.href,
+    }
+  })
+}
+
+if (COMPASS_TEST_DATA_LAKE_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas Data Lake'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_DATA_LAKE_URL,
+    }
+  })
+}
+
+if (COMPASS_TEST_SERVERLESS_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas Serverless'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_SERVERLESS_URL,
+    }
+  })
+}
+
+[
+  'enterprise',
+  'ldap',
+  'scramReadWriteAnyDatabase',
+  'scramReadWriteAnyDatabaseScramSha1',
+  'scramReadWriteAnyDatabaseScramSha256',
+  'scramOnlyScramSha1',
+  'scramOnlyScramSha256',
+  'scramEncodedPassword',
+  'scramPrivilegesOnNonExistingDatabases',
+  'scramPrivilegesOnNonExistingCollections',
+  'scramAlternateAuthDb',
+  'sharded',
+  'sshPassword',
+  'sshIdentityKey',
+  'sshIdentityKeyWithPassphrase',
+  'tlsUnvalidated',
+  'tlsServerValidation',
+  'tlsServerAndClientValidation',
+  'tlsServerAndClientValidationKeyCrt',
+  'tlsX509',
+  'tlsX509WithSsh',
+  'kerberos',
+  'kerberosAlternate',
+  'kerberosCrossRealm',
+].forEach(envName => connections.push({
+  favorite: {
+    name: envName,
+  },
+  connectionOptions: envs.getConnectionOptions(envName)
+}));
+
+async function main() {
+  const channels = {
+    dev: 'MongoDB Compass Dev',
+    beta: 'MongoDB Compass Beta',
+    stable: 'MongoDB Compass'
+  };
+
+  const channel = process.argv[2];
+  if (!Object.keys(channels).includes(channel)) {
+    throw new Error(`USAGE: electron scripts/import-test-connections.js ${Object.keys(channels).join("|")}`);
+  }
+
```

Comment:
Before running the import I'd maybe check if it's being run with local electron binary and warn / throw to avoid people running into "allow access" pop-up if they accidentally run it with node (but it's definitely just a follow-up improvement that we can do separately)

Reply:
Ah with node.js it already throws. But the "allow access" pop-up will always happen even with electron cause we are not setting the right appname etc.

If we want to make it work probably we could have `import-test-connections.js` and `import-test-connections-electron.js` with one calling the other with the right settings.

Reply:
Yeah, I think prompt like that is good enough. I wonder if we can make it work at least for dev without prompting though... Running it with electron directly without the need to do it manually also sounds pretty cool! Totally not a blocker though, I think it's already pretty cool to be able to import them with the script and the multiple prompts issue is only here until we do the change and use one cipher key for all connections anyway